src/lib/free-queue/test/free-queue.test.js

```diff
@@ -0,0 +1,162 @@
+import { expect } from 'https://cdnjs.cloudflare.com/ajax/libs/chai/5.1.1/chai.js';
+import { FreeQueue, MAX_CHANNEL_COUNT, RENDER_QUANTUM_FRAMES } from '../free-queue.js';
+
+// Mock WASM module
+const mockWasmModule = {
+  // Simulate memory allocation
+  _malloc: (size) => new ArrayBuffer(size), 
+  // Simulate memory deallocation
+  _free: () => {}, 
+  // Simulate HEAPF32
+  HEAPF32: new Float32Array(1024), 
+};
+
+describe('FreeQueue Class', () => {
+  const bufferLength = 512;
+  const channelCount = 2;
+  const maxChannelCount = 4;
+  let freeQueue = null;
+
+  beforeEach(() => {
+    freeQueue = new FreeQueue(mockWasmModule, bufferLength, channelCount, maxChannelCount);
+  });
+
+  afterEach(() => {
+    freeQueue.free();
+  });
+
+  describe('Initialization', () => {
+    it('should initialize with correct properties', () => {
+      expect(freeQueue.length).to.equal(bufferLength);
+      expect(freeQueue.numberOfChannels).to.equal(channelCount);
+      expect(freeQueue.maxChannelCount).to.equal(maxChannelCount);
+    });
+
+    it('should allocate the correct amount of memory', () => {
+      const dataByteSize = channelCount * bufferLength * Float32Array.BYTES_PER_ELEMENT;
+      expect(freeQueue.getPointer()).to.be.instanceof(ArrayBuffer);
+      expect(freeQueue.getPointer().byteLength).to.equal(dataByteSize);
+    });
+  });
+
+  describe('Channel Adaptation', () => {
```

Comment:
Can we port this test to the FreeQueueSAB test suite as well, or does that not allow a variable channel count?

Can we also add another test that tries maxChannelCount+1 and shows the channel count doesn't update / an error is thrown / etc.?

Reply:
I'll look into it and let you know!